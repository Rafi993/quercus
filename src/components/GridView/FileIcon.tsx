import React from 'react';

import Icon from '../Icon';

interface Props {
  type: string;
}

const FileIcon: React.FC<Props> = ({ type }) => {
  switch (type) {
    case 'png':
    case 'jpg':
    case 'webp':
      return <Icon type="image" />;
    case 'mp4':
    case 'avi':
      return <Icon type="video" />;
    case 'pdf':
      return <Icon type="pdf" />;
    default:
      return <Icon type="file" />;
  }
};

export default FileIcon;
