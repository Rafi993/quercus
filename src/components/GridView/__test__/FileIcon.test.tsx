import React from 'react';
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import FileIcon from '../FileIcon';

afterEach(cleanup);

describe('File icon is rendered correctly', () => {
  it('renders image icon', () => {
    const { asFragment } = render(<FileIcon type="jpg" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders video icon', () => {
    const { asFragment } = render(<FileIcon type="mp4" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders file icon', () => {
    const { asFragment } = render(<FileIcon type="xkcd" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
