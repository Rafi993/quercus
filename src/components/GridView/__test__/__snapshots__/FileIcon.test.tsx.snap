// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File icon is rendered correctly renders file icon 1`] = `
<DocumentFragment>
  <svg
    role="img"
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm64 236c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-64c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12v8zm0-72v8c0 6.6-5.4 12-12 12H108c-6.6 0-12-5.4-12-12v-8c0-6.6 5.4-12 12-12h168c6.6 0 12 5.4 12 12zm96-114.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
      fill="#888"
    />
  </svg>
</DocumentFragment>
`;

exports[`File icon is rendered correctly renders image icon 1`] = `
<DocumentFragment>
  <svg
    role="img"
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M384 121.941V128H256V0h6.059a24 24 0 0 1 16.97 7.029l97.941 97.941a24.002 24.002 0 0 1 7.03 16.971zM248 160c-13.2 0-24-10.8-24-24V0H24C10.745 0 0 10.745 0 24v464c0 13.255 10.745 24 24 24h336c13.255 0 24-10.745 24-24V160H248zm-135.455 16c26.51 0 48 21.49 48 48s-21.49 48-48 48-48-21.49-48-48 21.491-48 48-48zm208 240h-256l.485-48.485L104.545 328c4.686-4.686 11.799-4.201 16.485.485L160.545 368 264.06 264.485c4.686-4.686 12.284-4.686 16.971 0L320.545 304v112z"
      fill="#888"
    />
  </svg>
</DocumentFragment>
`;

exports[`File icon is rendered correctly renders video icon 1`] = `
<DocumentFragment>
  <svg
    role="img"
    viewBox="0 0 384 512"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M369.941 97.941l-83.882-83.882A48 48 0 0 0 252.118 0H48C21.49 0 0 21.49 0 48v416c0 26.51 21.49 48 48 48h288c26.51 0 48-21.49 48-48V131.882a48 48 0 0 0-14.059-33.941zM332.118 128H256V51.882L332.118 128zM48 464V48h160v104c0 13.255 10.745 24 24 24h104v288H48zm228.687-211.303L224 305.374V268c0-11.046-8.954-20-20-20H100c-11.046 0-20 8.954-20 20v104c0 11.046 8.954 20 20 20h104c11.046 0 20-8.954 20-20v-37.374l52.687 52.674C286.704 397.318 304 390.28 304 375.986V264.011c0-14.311-17.309-21.319-27.313-11.314z"
      fill="#888"
    />
  </svg>
</DocumentFragment>
`;
